randovania/gui/preset_settings/preset_transporter_tab.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+
+import dataclasses
+from typing import TYPE_CHECKING
+
+from PySide6 import QtCore, QtWidgets
+
+from randovania.gui.lib import signal_handling
+from randovania.gui.lib.node_list_helper import NodeListHelper
+from randovania.gui.preset_settings.preset_tab import PresetTab
+from randovania.layout.lib.teleporters import (
+    TeleporterList,
+    TeleporterTargetList,
+)
+from randovania.patching.prime import elevators
+
+if TYPE_CHECKING:
+    from randovania.game_description.db.area_identifier import AreaIdentifier
+    from randovania.game_description.db.node_identifier import NodeIdentifier
+    from randovania.game_description.game_description import GameDescription
+    from randovania.gui.lib.scroll_protected import ScrollProtectedComboBox
+    from randovania.gui.lib.window_manager import WindowManager
+    from randovania.interface_common.preset_editor import PresetEditor
+
+
+class PresetTransporterTab(PresetTab, NodeListHelper):
+    _teleporters_source_for_location: dict[NodeIdentifier, QtWidgets.QCheckBox]
+    _teleporters_target_for_region: dict[str, QtWidgets.QCheckBox]
+    _teleporters_target_for_area: dict[AreaIdentifier, QtWidgets.QCheckBox]
+    _teleporters_target_for_node: dict[NodeIdentifier, QtWidgets.QCheckBox]
+    compatible_modes = []
+    teleporters_layout: QtWidgets.QVBoxLayout
+    teleporters_combo: ScrollProtectedComboBox
+    teleporters_description_label: QtWidgets.QLabel
+    teleporters_source_group: QtWidgets.QGroupBox
+    teleporters_target_group: QtWidgets.QGroupBox
+    teleporters_target_layout: QtWidgets.QGridLayout
+
+    def __init__(self, editor: PresetEditor, game_description: GameDescription, window_manager: WindowManager):
```

Comment:
If the class fields are named teleporters, then the class name should also be Teleporter

Reply:
Yeah, see the last point of my last message: If no one is opposed, I will call it always teleporter internal for all games and any more fitting name outside for the UI and presentation to the user.